src/Nethermind/Nethermind.Core/Caching/ClockCache.cs

```diff
@@ -154,7 +157,7 @@ public bool Delete(TKey key)
         using var lockRelease = _lock.Acquire();
 
         base.Clear();
-        _cacheMap.Clear();
+        _cacheMap.NoResizeClear();
```

Comment:
maybe have a `NoLockNoResizeClear` - as you don't need the `ConcurentDictionaryLock` here?

Reply:
Still do or it could have cleared one array but not the other and mess up a reader

Reply:
The reader side isn't under lock

Reply:
ok you convinced me.